lib/generators/declare_schema/migration/migration_generator.rb

```diff
@@ -67,51 +65,55 @@ def migrate
       say down
       say "----------------------------------"
 
-      action = options[:generate] && 'g' ||
-               options[:migrate] && 'm' ||
-               choose("\nWhat now: [g]enerate migration, generate and [m]igrate now or [c]ancel?", /^(g|m|c)$/)
-
-      if action != 'c'
-        if name.blank? && !options[:default_name]
-          final_migration_name = choose("\nMigration filename: [<enter>=#{migration_name}|<custom_name>]:", /^[a-z0-9_ ]*$/, migration_name).strip.gsub(' ', '_')
-        end
-        final_migration_name = migration_name if final_migration_name.blank?
-
-        up.gsub!("\n", "\n    ")
-        up.gsub!(/ +\n/, "\n")
-        down.gsub!("\n", "\n    ")
-        down.gsub!(/ +\n/, "\n")
-
-        @up = up
-        @down = down
-        @migration_class_name = final_migration_name.camelize
-
-        migration_template('migration.rb.erb', "db/migrate/#{final_migration_name.underscore}.rb")
-        if action == 'm'
-          case Rails::VERSION::MAJOR
-          when 4
-            rake('db:migrate')
-          else
-            rails_command('db:migrate')
-          end
+      final_migration_name =
+        name.presence ||
+        if !options[:default_name]
+          choose("\nMigration filename (spaces will be converted to _) [#{default_migration_name}]:", /^[a-z0-9_ ]*$/,
+                 default_migration_name).strip.gsub(' ', '_').presence
+        end ||
+        default_migration_name
+
+      @up = indent(up, 4)
+      @down = indent(down, 4)
+      @migration_class_name = final_migration_name.camelize
+
+      migration_template('migration.rb.erb', "db/migrate/#{final_migration_name.underscore}.rb")
+
+      db_migrate_command = ::DeclareSchema.db_migrate_command
+      if options[:migrate]
+        say db_migrate_command
```

Comment:
`say`ing the migration command before we strip the `bundle exec rails|rake` seems like it might cause some confusion down the line. If we want this sort of logging, what about moving it to the `rake` and `rails_command` methods?

Reply:
My reasoning is that you need the `bundle exec rails|rake` if you run from the command line...but internally, that prefix has already been run so we don't need to (aren't allowed to, in fact) repeat it.